dp/generate_climos.py

```diff
@@ -39,6 +39,86 @@
 cdo = Cdo()
 
 
+def has_climo_periods(input_file, climo):
+    '''
+    This function checks if the given input NetCDF dataset has a variable "climo_periods".
+    generate_climos requires the input dataset to have the variable, thus this function will
+    warn if there is no such variable.
+    '''
+    periods = input_file.climo_periods.keys() & climo
+    logger.info("climo_periods: {}".format(periods))
+    if len(periods) == 0:
+        logger.info(f"{input_file.filepath()} has no variable 'climo_periods'")
```

Comment:
> Will there actually be any output without this?

I believe there is no output if the given input does not have climo periods.

> Also, do we want to continue the script if we have no climo periods? 

@rod-glover what do you think? Should the program halt here?

Reply:
> I believe there is no output if the given input does not have climo periods.

We can wait to hear what Rod says but I think if there is no output then there is no reason to continue. That's just my $0.02 though.

Reply:
You can bail here if you want, or you can just let the logic of the code do its thing and produce nothing more. In general, I favour the early-return style because it tells you where you can stop thinking (always good for me :) ). Not so sure here as bailing is, I think, a call to `sys.exit()`, which is much more problematic than a `return`. However, with the refactoring I suggest, that call to `sys.exit()` may end up at the very top level, which would then be OK.

TL;DR: Let's wait till you refactor and look again. Probably bail, though.